cmd/protecodeExecuteScan.go

```diff
@@ -0,0 +1,317 @@
+
+	jsonData, _ := json.Marshal(result)
+	ioutil.WriteFile("Vulns.json", jsonData, 0644)
+
+	//check if result is ok else notify
+	if len(result.Result.Status) > 0 && result.Result.Status == "F" {
+		log.Entry().Fatal("Protecode scan failed, please check the log and protecode backend for more details.")
+	}
+	//loadReport
+	if config.Verbose {
+		log.Entry().Info("Protecode scan debug, load report")
+	}
+	resp := client.LoadReport(config.ReportFileName, productID)
+
+	//save report to filesystem
+	err := writeReportToFile(*resp, config.ReportFileName)
+	if err != nil {
+		return parsedResult, productID
+	}
+	//clean scan from server
+	if config.Verbose {
+		log.Entry().Info("Protecode scan debug, delete scan")
+	}
+	client.DeleteScan(config.CleanupMode, productID)
+
+	//count vulnerabilities
+	if config.Verbose {
+		log.Entry().Info("Protecode scan debug, parse result")
+	}
+	parsedResult = client.ParseResultForInflux(result.Result, config.ProtecodeExcludeCVEs)
+
+	return parsedResult, productID
+}
+
+func setInfluxData(influx *protecodeExecuteScanInflux, result map[string]int) {
+
+	influx.protecodeData.fields.historicalVulnerabilities = fmt.Sprintf("%v", result["historical_vulnerabilities"])
+	influx.protecodeData.fields.triagedVulnerabilities = fmt.Sprintf("%v", result["triaged_vulnerabilities"])
+	influx.protecodeData.fields.excludedVulnerabilities = fmt.Sprintf("%v", result["excluded_vulnerabilities"])
+	influx.protecodeData.fields.minorVulnerabilities = fmt.Sprintf("%v", result["minor_vulnerabilities"])
+	influx.protecodeData.fields.majorVulnerabilities = fmt.Sprintf("%v", result["major_vulnerabilities"])
+	influx.protecodeData.fields.vulnerabilities = fmt.Sprintf("%v", result["vulnerabilities"])
+}
+
+func writeReportDataToJSONFile(config *protecodeExecuteScanOptions, result map[string]int, productID int, writeToFile func(f string, d []byte, p os.FileMode) error) {
+
+	protecodeData := protecodeData{}
+	protecodeData.ProtecodeServerURL = config.ProtecodeServerURL
+	protecodeData.ProtecodeFailOnSevereVulnerabilities = config.ProtecodeFailOnSevereVulnerabilities
+	protecodeData.ProtecodeExcludeCVEs = config.ProtecodeExcludeCVEs
+	protecodeData.Target = config.ReportFileName
+	protecodeData.Mandatory = true
+	protecodeData.ProductID = fmt.Sprintf("%v", productID)
+	protecodeData.Count = fmt.Sprintf("%v", result["count"])
+	protecodeData.Cvss2GreaterOrEqualSeven = fmt.Sprintf("%v", result["cvss2GreaterOrEqualSeven"])
+	protecodeData.Cvss3GreaterOrEqualSeven = fmt.Sprintf("%v", result["cvss3GreaterOrEqualSeven"])
+	protecodeData.ExcludedVulnerabilities = fmt.Sprintf("%v", result["excluded_vulnerabilities"])
+	protecodeData.TriagedVulnerabilities = fmt.Sprintf("%v", result["triaged_vulnerabilities"])
+	protecodeData.HistoricalVulnerabilities = fmt.Sprintf("%v", result["historical_vulnerabilities"])
+
+	jsonData, _ := json.Marshal(protecodeData)
+
+	writeToFile("report.json", jsonData, 0644)
```

Comment:
naming? should we add the stepName to the name of the report file or does it need to have this name?

Reply:
Valid point. Perhaps not just step name but somehow also the module to avoid clashes in parallel execution scenarios.